### PR TITLE
Handle uint overflow for segment id in replays processor

### PR DESCRIPTION
**Pull request by volokluev**
This is causing too much noise on replays consumer.

### CODE REVIEW

**Review by MeredithAnya — APPROVED** [review]

File: rust_snuba/src/processors/replays.rs [hunk 1]
```diff
@@ -144,14 +155,14 @@ pub fn deserialize_message(
...
                 sdk_version: event.sdk.version.unwrap_or_default(),
-                segment_id: event.segment_id,
                 timestamp: event.timestamp as u32,
                 trace_ids: event.trace_ids.unwrap_or_default(),
                 urls: event.urls.unwrap_or_default(),
                 user,
                 user_email: event.user.email.unwrap_or_default(),
                 user_id: user_id.unwrap_or_default(),
                 user_name: event.user.username.unwrap_or_default(),
+                segment_id,
```
Comment (phacops) [inline rust_snuba/src/processors/replays.rs:165]: There's always a Rust 1-liner in case you need it.
```suggestion
+                segment_id: event.segment_id.filter(|s| *s <= (u16::MAX as u32)).map_or(None, |s| Some(s as u16)),
```

**Review by phacops — APPROVED** [review]

File: rust_snuba/src/processors/replays.rs [hunk 2]
```diff
@@ -115,6 +115,17 @@ pub fn deserialize_message(
...
+            let segment_id = match event.segment_id {
+                Some(s_id) => {
+                    if s_id <= u16::MAX.into() {
+                        Some(s_id as u16)
+                    } else {
+                        None
+                    }
+                }
+                None => None,
+            };
```
Comment (onkar) [inline rust_snuba/src/processors/replays.rs:127]:
```
let segment_id = event.segment_id.and_then(|s_id| {
    if s_id <= u16::MAX as u32 {
        Some(s_id as u16)
    } else {
        None
    }
});
```

The` into()` conversion is unnecessary because `u16::MAX` is already a `u16`. `and_then` simplifies the option handling.

File: rust_snuba/src/processors/replays.rs [hunk 3]
```diff
@@ -469,6 +480,77 @@ mod tests {
...
+                "segment_id": null,
+                "start_time": 100,
+                "type": "replay_event"
+            }}"#
+        );
+
+        let (rows, _) = deserialize_message(data.as_bytes(), 0, 0).unwrap();
+        let replay_row = rows.first().unwrap();
+        assert_eq!(replay_row.segment_id, None);
+    }
```
Comment (onkar) [inline rust_snuba/src/processors/replays.rs:552]:
```
let valid_segment_id = u16::MAX as u32;  // Maximum valid value
    let payload_valid = payload.replace("4206942069", &valid_segment_id.to_string());
    let data_valid = format!(
        r#"{{
            "payload": {payload_valid:?},
            "project_id": 1,
            "replay_id": "048aa04be40243948eb3b57089c519ee",
            "retention_days": 30,
            "segment_id": null,
            "start_time": 100,
            "type": "replay_event"
        }}"#
    );

    let (rows_valid, _) = deserialize_message(data_valid.as_bytes(), 0, 0).unwrap();
    let replay_row_valid = rows_valid.first().unwrap();

    // segment_id should be correctly parsed when it's within u16 range
    assert_eq!(
        replay_row_valid.segment_id,
        Some(u16::MAX),
        "segment_id should be stored when within u16 range"
    );
```

nit: Consider adding a test when the segment id is set to the maximum value.

**Review by onkar — APPROVED** [review]
LGTM, suggested a couple of things.
